### PR TITLE
chore: add option for changelog to write to file

### PR DESCRIPTION
**Pull request by lukekarrys**
This makes our changelog script do more stuff for us! 🤖

**tl;dr**
- By default you can run it with no parameters and it will figure out the most recent tag
- Will error if there are unpushed commits or you are on the wrong branch
- Uses the changelog types directly from our `@npmcli/template-oss` config
- You can run it with an old tag to see how it would generate the notes for that release. This is helpful as we iterate on the script we can see how it would perform historically.
- a `--write` flag which will write the notes directly to the `CHANGELOG.md` file in the correct place. And if you run this multiple times, it will find the existing section and update it in place.
- a `--read` flag to get the release notes for a version directly from the file. This is a convenience method to get the notes after manually editing them instead of copy/pasting

### CODE REVIEW

File: scripts/changelog.js [hunk 1]
```diff
@@ -1,44 +1,180 @@
...
+    process.exit(0)
+  }
+
+  return args
+}
 
 const parseArgs = (argv) => {
   const result = {
+    file: resolve(__dirname, '..', 'CHANGELOG.md'),
+    branch: 'latest',
```
Comment (wraithgar) [inline scripts/changelog.js:125]: I think this was originally `origin/latest` in an attempt to solve the "my local git differs from github" problem.  It obviously doesn't, but perhaps there is a way for us to have this guard when building the changelog?

**Review by wraithgar — APPROVED** [review]

File: scripts/changelog.js (hunk 1 above)
Comment (lukekarrys) [reply scripts/changelog.js:125]: This function calls `assertArgs` at the end which will throw an error if you're not on `latest` or you have unpushed commits. This made me realize we should also add a check for the opposite: remote has commits that you dont.
